### PR TITLE
Head

### PR DESCRIPTION
**Pull request by wotzisname**
Cdata sections are not being output in the toString method. This can break the XML

### CODE REVIEW

File: lib/xmldoc.js [hunk 1]
```diff
@@ -139,8 +143,12 @@ XmlElement.prototype.toStringWithIndent = function(indent, options) {
     if (Object.prototype.hasOwnProperty.call(this.attr, name))
         s += " " + name + '="' + escapeXML(this.attr[name]) + '"';
 
-  var finalVal = escapeXML(this.val.trim());
-
+  var finalVal = '';
+  if (this.isValCdata){
+    finalVal = '<![CDATA['+this.val+']]>';
+  } else {
+    escapeXML(this.val);
```
Comment (nfarina) [inline lib/xmldoc.js]: This needs to be assigned to finalVal, otherwise finalVal is left empty

**Comment by wotzisname** [comment]
Update to correctly assign value.

File: lib/xmldoc.js [hunk 2]
```diff
@@ -139,8 +143,12 @@ XmlElement.prototype.toStringWithIndent = function(indent, options) {
     if (Object.prototype.hasOwnProperty.call(this.attr, name))
         s += " " + name + '="' + escapeXML(this.attr[name]) + '"';
 
-  var finalVal = escapeXML(this.val.trim());
-
+  var finalVal = '';
+  if (this.isValCdata){
+    finalVal = '<![CDATA['+this.val+']]>';
+  } else {
+    finalVal = escapeXML(this.val);
```
Comment (nfarina) [inline lib/xmldoc.js]: I forget if I commented on this before but this has lost the `trim()` from the previous implementation.

**Comment by wotzisname** [comment]
Why would you include a trim()? Surely it's not derived behavior to randonly trim spaces from the start and end of attributes. Is there something I'm missing in the Xml Spec?

**Comment by nfarina** [comment]
This method is not intended to produce strictly valid XML, and definitely does not promise to return even exactly what you gave it. It is for debugging only, and if you find it useful for other things that's great. The original implementation uses trim to produce a compact string for debugging purposes, so unless there's a good reason to change it then let's leave it as it was.

**Comment by wotzisname** [comment]
Ah okay. This just seems to be the best cross platform Xml library for Node to manipulate and save xml. If the spaces were in the original value it seems strange to remove them - when would there be more than a few? We need to manipulate the xml and re-save it so maybe we should create a fork of this library if that's not the direction that you want to go in. The fact that there's been another pull request to implement CData seems to suggest people find your library useful for more than just debugging, but obviously it's your perogative.

**Comment by nfarina** [comment]
Yeah I have had this discussion a lot it seems :) My primary goal of the library is for it to be as simple and small as possible. If it doesn't add too much code, perhaps a flag could be added to preserve whitespace?

**Comment by wotzisname** [comment]
I've added back the trim and added an option to preserve the spaces around the value. I hope this works for you now.

**Comment by nfarina** [comment]
Looks great! Thanks.

**Comment by wotzisname** [comment]
I was wondering if/when you're planning the next version release? We need the changes we made and it's always nice to be running off an "official" version.
Thanks

**Comment by nfarina** [comment]
Just published.

**Pull request merged**